### PR TITLE
[ICDS Dashboard] Fix for District not available in Dropdown in CAS Data Exports

### PR DESCRIPTION
**Pull request by challabeehyv**
https://app.asana.com/0/1185205808600689/1186409062810626/f

##### SUMMARY
Fix for District not available in Dropdown in CAS Data Exports

##### FEATURE FLAG
No

### CODE REVIEW

File: custom/icds_reports/static/js/directives/cas-export/cas-export.directive.js
Comment (ShubhamBansal1997) [inline custom/icds_reports/static/js/directives/cas-export/cas-export.directive.js:144]: Do you want to check null and undefined? if yes then doing it that way would be better ```vm.userLocationId === null || vm.userLocationId === undefined```

**Review by ShubhamBansal1997 — COMMENTED** [review]
Left one comment, good to test on staging before merging

File: custom/icds_reports/static/js/directives/cas-export/cas-export.directive.js
Comment (challabeehyv) [reply custom/icds_reports/static/js/directives/cas-export/cas-export.directive.js:144]: following the convention from download directive

Comment (ShubhamBansal1997) [reply custom/icds_reports/static/js/directives/cas-export/cas-export.directive.js:144]: can you point out the same? because if you are doing the former it might resolve the same as string not as actual null or undefined.

Comment (challabeehyv) [reply custom/icds_reports/static/js/directives/cas-export/cas-export.directive.js:144]: https://github.com/challabeehyv/commcare-hq/blob/cr/bf_birth/custom/icds_reports/static/js/directives/download/download.directive.js#L227

They have deliberately put it that way which means the string values are coming as null or undefined. so thought of leaving it like that since both uses same method

Comment (ShubhamBansal1997) [reply custom/icds_reports/static/js/directives/cas-export/cas-export.directive.js:144]: Sounds weird tagging @czue if he got some insight about this

Comment (calellowitz) [reply custom/icds_reports/static/js/directives/cas-export/cas-export.directive.js:144]: agree with shubham that this is a little odd (also seems slightly different unless its converting `null` and `undefined` to strings silently somewhere), but if this is the convention elsewhere its probably fine

File: custom/icds_reports/static/js/directives/cas-export/cas-export.directive.js
Comment (calellowitz) [inline custom/icds_reports/static/js/directives/cas-export/cas-export.directive.js:133]: it might be clearer to rename this to something like `userHaveAccessToLevel` or `accessToAllLocationsForLevel` or something that makes clear its only one level, and distinguishes it from the very similarly named variable that already exists and is used in the same conditional below

File: custom/icds_reports/static/js/directives/cas-export/cas-export.directive.js
Comment (calellowitz) [inline custom/icds_reports/static/js/directives/cas-export/cas-export.directive.js:148]: it it possible for `haveAccessToAllLocations` to be true while `vm.userHaveAccessToAllLocations(locations)` is not true? it seems like the second bit is redundant. 
Also, i think you have an unnecessary level or parens

**Review by calellowitz — CHANGES_REQUESTED** [review]

File: custom/icds_reports/static/js/directives/cas-export/cas-export.directive.js
Comment (challabeehyv) [reply custom/icds_reports/static/js/directives/cas-export/cas-export.directive.js]: done

File: custom/icds_reports/static/js/directives/cas-export/cas-export.directive.js
Comment (challabeehyv) [reply custom/icds_reports/static/js/directives/cas-export/cas-export.directive.js]: makes sense. refactored

File: custom/icds_reports/static/js/directives/cas-export/cas-export.directive.js
Comment (czue) [reply custom/icds_reports/static/js/directives/cas-export/cas-export.directive.js:143]: I don't have any familiarity with this pattern or why it would be used. Looks like @Rohit25negi is the author of the original code that was used - maybe he has some insight? https://github.com/challabeehyv/commcare-hq/commit/e8e9a69b0a844a60d8d8f8a052467f3115269030#diff-57a6e3501ebaf4aff43ac70619c535a6R207-R209

Comment (Rohit25negi) [reply custom/icds_reports/static/js/directives/cas-export/cas-export.directive.js:143]: Yeah, I touched them when working on the Tabular report location filters. But only used what was being used. Not super confident about when it was started and where it may be silently converting them to the strings.
I think its the pattern other places we have been following. fine to keep it. Would be great if to also include actual `null` and `undefined` in the list

**Review by Rohit25negi — COMMENTED** [review]
overall looks good. left one comment

**Comment by challabeehyv** [comment]
> overall looks good. left one comment

added

File: custom/icds_reports/static/js/directives/cas-export/cas-export.directive.js
Comment (challabeehyv) [reply custom/icds_reports/static/js/directives/cas-export/cas-export.directive.js]: added

**Review by Rohit25negi — APPROVED** [review]

**Comment by Rohit25negi** [comment]
Looks good to me. Will wait for @calellowitz  to have another review as some change requests were made by him.

**Comment by challabeehyv** [comment]
closing this as duplicate to https://github.com/dimagi/commcare-icds/pull/59

**Pull request closed without merging**
